src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -599,6 +599,47 @@ export class StudyViewPageStore
         );
     }
 
+    @action.bound
+    changePreviewDimension(uniqueKey: string, previewShown: boolean): void {
+        const dimension = previewShown
+            ? STUDY_VIEW_CONFIG.layout.dimensions[
+                  ChartTypeEnum.BAR_PREVIEW_CHART
+              ]
+            : STUDY_VIEW_CONFIG.layout.dimensions[ChartTypeEnum.BAR_CHART];
+        this.chartsDimension.set(uniqueKey, dimension);
+    }
+
+    @action
+    togglePreview = (uniqueKey: string): void => {
+        if (this._customDataBinFilterSet.has(uniqueKey)) {
+            const filterSet = this._customDataBinFilterSet.get(uniqueKey);
+            const newFilter = _.clone(filterSet!);
+            newFilter.showPreview = !this.isPreviewShown(uniqueKey);
+            this.changePreviewDimension(uniqueKey, newFilter.showPreview);
+            this._customDataBinFilterSet.set(uniqueKey, newFilter as any);
+        } else {
+            const filterSet = this.getDataBinFilterSet(uniqueKey);
+            const newFilter = _.clone(filterSet.get(uniqueKey)!);
+            newFilter.showPreview = !this.isPreviewShown(uniqueKey);
+            this.changePreviewDimension(uniqueKey, newFilter.showPreview);
+            filterSet.set(uniqueKey, newFilter as any);
+        }
+    };
+
+    public isPreviewShown = (uniqueKey: string): boolean => {
+        if (this._customDataBinFilterSet.has(uniqueKey)) {
+            const filter = this._customDataBinFilterSet.get(uniqueKey);
+            return filter === undefined
+                ? false
+                : (filter!.showPreview as boolean);
+        } else {
+            const filter = this.getDataBinFilterSet(uniqueKey).get(uniqueKey)!;
+            return filter?.showPreview === undefined
+                ? false
+                : (filter!.showPreview as boolean);
```

Comment:
i think this could be simplified. one way could be to use ternary operator (n ? x : y) to get the filter and something like `return filter?.showPreview || false` for the return value